From 3093bceb003846aef30b47bc522fcef2fc676b70 Mon Sep 17 00:00:00 2001
From: Harald Sitter <sitter@kde.org>
Date: Fri, 16 Oct 2026 00:08:51 +0200
Subject: [PATCH 1/2] systemd: decouple baloo from our dependency tree

.target files are special in that they automatically add implicit After=
relationships for Wants= relationships. this then leads to a tidy
dependency loop because baloo wants to start up alongside
graphical-session.target which in turn is After plasma-workspace. to
break this loop we explicitly order workspace before baloo, thus
disabling the automatic after rule
---
 startkde/systemd/plasma-workspace.target | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/startkde/systemd/plasma-workspace.target b/startkde/systemd/plasma-workspace.target
index 52424ae1c1..25c0c17b81 100644
--- a/startkde/systemd/plasma-workspace.target
+++ b/startkde/systemd/plasma-workspace.target
@@ -15,9 +15,15 @@ Wants=plasma-kaccess.service
 Wants=plasma-startupsound.service
 Wants=xdg-desktop-autostart.target
 BindsTo=graphical-session.target
+
+# .target units automatically add After= for Wants=. In some cases we actually do not want there to be a dependency order.
+# Explicitly setting a Before= disables the automatic After= that would normally be added.
+# This in particular avoids dependency loops.
 Before=graphical-session.target
 Before=xdg-desktop-autostart.target
 Before=plasma-startupsound.service
 Before=plasma-restoresession.service
+Before=kde-baloo.service
+
 RefuseManualStart=yes
 StopWhenUnneeded=true
-- 
GitLab


From 1113aae73ba0dab3f98bfd0ca3a6e0ca9eafd6e1 Mon Sep 17 00:00:00 2001
From: Harald Sitter <sitter@kde.org>
Date: Fri, 16 Oct 2026 10:56:11 +0200
Subject: [PATCH 2/2] systemd: remove the relationships with kde-baloo.service

somewhat unfortunately we've accumulated Wants relationships that really
should have been passive WantedBy from the other side. let's start
cleaning things up

kde-baloo.service already declares itself
WantedBy=graphical-session.target. As such it will get dragged into the
startup tree. our relationship in the other directly only served to get
us into a dependency loop without having any benefit
---
 startkde/systemd/plasma-workspace.target | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/startkde/systemd/plasma-workspace.target b/startkde/systemd/plasma-workspace.target
index 25c0c17b81..49356b97e0 100644
--- a/startkde/systemd/plasma-workspace.target
+++ b/startkde/systemd/plasma-workspace.target
@@ -2,19 +2,20 @@
 Description=KDE Plasma Workspace
 Requires=plasma-core.target
 Requires=graphical-session.target
+BindsTo=graphical-session.target
+
+# Be mindful of what we Wants=. Usually it makes more sense to get passively WantedBy= rather than actively Wants=.
 Wants=plasma-restoresession.service
 Wants=plasma-xembedsniproxy.service
 Wants=plasma-gmenudbusmenuproxy.service
 Wants=plasma-powerdevil.service
 Wants=plasma-polkit-agent.service
-Wants=kde-baloo.service
 Wants=plasma-foreground-booster.service
 Wants=plasma-kwallet-pam.service
 Wants=plasma-kglobalaccel.service
 Wants=plasma-kaccess.service
 Wants=plasma-startupsound.service
 Wants=xdg-desktop-autostart.target
-BindsTo=graphical-session.target
 
 # .target units automatically add After= for Wants=. In some cases we actually do not want there to be a dependency order.
 # Explicitly setting a Before= disables the automatic After= that would normally be added.
@@ -23,7 +24,6 @@ Before=graphical-session.target
 Before=xdg-desktop-autostart.target
 Before=plasma-startupsound.service
 Before=plasma-restoresession.service
-Before=kde-baloo.service
 
 RefuseManualStart=yes
 StopWhenUnneeded=true
-- 
GitLab

